src/main/java/seedu/address/logic/parser/ModCommandParser.java

```diff
@@ -0,0 +1,127 @@
+     */
+    @Override
+    public ModCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+
+        final Matcher matcher = BASIC_COMMAND_FORMAT.matcher(args.trim());
+        String trimmedArgs = args.trim();
+        // check if empty
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, ModCommand.MESSAGE_USAGE));
+        }
+        if (!matcher.matches()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, ModCommand.MESSAGE_USAGE));
+        }
+        // parse
+        final String commandWord = matcher.group("commandWord");
+        final String arguments = matcher.group("arguments");
+
+        // TODO: Add cases for other mod commands
+        switch (commandWord) {
+        case ModAddCommand.COMMAND_WORD:
+            return parseAddCommand(arguments);
+        default:
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, ModCommand.MESSAGE_USAGE));
+        }
+    }
+
+    private ModAddCommand parseAddCommand(String args) throws ParseException {
+        Index index;
+        args = args.trim();
+        String indexFromCommand = getIndexFromCommand(args);
+        Set<String> modsFromCommand = getModsFromCommand(args);
+        Optional<Set<Mod>> mods = parseMods(modsFromCommand);
+        if (mods.isEmpty()) {
+            throw new ParseException(ModCommand.MESSAGE_MODS_EMPTY);
+        }
+
+        try {
+            index = ParserUtil.parseIndex(indexFromCommand);
+        } catch (ParseException pe) {
+            throw new ParseException(ModCommand.MESSAGE_INDEX_INVALID, pe);
+        }
+        return new ModAddCommand(index, mods.get());
+    }
+
+    /**
+     * Converts a collection of strings representing mod names to a set of mods.
+     *
+     * @param mods A collection of mods in string.
+     * @return A set of mods.
+     * @throws ParseException When mod names are of incorrect format.
+     */
+    private Optional<Set<Mod>> parseMods(Collection<String> mods) throws ParseException {
+        assert mods != null;
+
+        if (mods.isEmpty()) {
+            return Optional.empty();
+        }
+        Collection<String> modSet = mods.size() == 1 && mods.contains("") ? Collections.emptySet() : mods;
+        return Optional.of(ParserUtil.parseMods(modSet));
+    }
+
+    /**
+     * Extracts out the index of the person specified in the user command.
+     *
+     * @param args The user command.
+     * @return The index of the person in String.
+     */
+    private String getIndexFromCommand(String args) throws ParseException {
+        String[] splittedArgs = args.split(" ");
+        String index = splittedArgs[0];
+        final Matcher matcher = INDEX_FORMAT.matcher(index.trim());
+        if (!matcher.matches()) {
+            throw new ParseException(ModCommand.MESSAGE_INDEX_EMPTY);
```

Comment:
When I enter `mod add -1 hi` which contains a negative index, "Index cannot be empty!" is shown but the index is not really empty.

For other commands like `mod add 10 cs2100` (invalid index) and `mod add 1` (empty mod), the displayed messages are good. 👍 

Perhaps we can do it like the `edit` command, i.e. for other situations not covered by invalid index (exceeds range), empty index or empty mod, show a general error message.
e.g. `edit -1`, `edit -1 n/ca` triggers the following message:
```
Invalid command format! 
edit: Edits the details of the person identified by the index number used in the displayed person list. Existing values will be overwritten by the input values.
Parameters: INDEX (must be a positive integer) [n/NAME] [t/TELEGRAM] [g/GITHUB] [p/PHONE] [e/EMAIL] [tag/TAG]...
Example: edit 1 g/john_doe p/91234567 e/johndoe@example.com
```